atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -159,14 +159,18 @@ def __init__(self, tasker, workflow,
             else:
                 self.log.info("No koji task")
 
+        self.original_koji_task_id = None
         try:
             metadata = get_build_json().get("metadata", {})
-            self.original_koji_task_id = int(metadata['labels']['original-koji-task-id'])
+            original_koji_task_id = metadata.get('labels', {}).get('original-koji-task-id')
         except Exception:
-            self.log.exception("Failed to fetch original koji task ID")
-            self.original_koji_task_id = None
+            self.log.info("Failed to fetch original koji task ID")
         else:
-            self.log.info("original Koji task ID: %s", self.original_koji_task_id)
+            if original_koji_task_id:
+                self.original_koji_task_id = int(original_koji_task_id)
```

Comment:
@SalatskySal 

Moving `int()` call here changes the original behavior. The change above in try-except is

```python
try:
    metadata = get_build_json().get("metadata", {})
    original_koji_task_id = metadata.get('labels', {}).get('original-koji-task-id')
except Exception:
    self.log.info("Failed to fetch original koji task ID")
```

My understand of the `get_build_json()` call is that the exit_sendmail plugin should run as one step through the container build workflow, there should be no use case to run this plugin itself only. Thus, through the workflow, `get_build_json` should be called already in the previous plugins before exit_sendmail runs. If no `$BUILD`, the build will fail earlier.

So, that two lines inside try-except does not raise any error in this context. The original code also catches potential error raised from `int()`, but this new change doesn't.

IMO, Robert's suggestion is a safe way. It keeps the change in a small scope and the original behavior, and does not introduce any new issue potentially.

Just FYI. :)

Reply:
@tkdchen thanks for the great explanation! It makes sense